### PR TITLE
Remove unused MQ related code

### PR DESCRIPTION
**Pull request by Glutexo**
Removed a few pieces of message queue related code that are no longer in use. These include:

- Mock producers (_NullEventProducer_) only logging messages instead of actually producing them. Added _INFO/DEBUG_ logging to the real producer instead.
- System Profile consumer task and its metrics. This service has been entirely replaced by Ingress.
- Configuration options controlling the tasks and event production. No need to have _another_ way to disable this one specific part of the application.

I didn’t do any further refactoring, although there is still a lot of work to do. When its time comes, it’s at least going to be easier.

No JIRA for this. Induced by #633.

### CODE REVIEW

**Comment by jharting** [comment]
@Glutexo should we remove this one also? https://github.com/RedHatInsights/insights-host-inventory/blob/master/tasks/__init__.py#L18

File: tasks/__init__.py [hunk 1]
```diff
@@ -33,26 +25,28 @@ def init_tasks(config, flask_app=None):
...
-    else:
-        logger.info("Starting NullProducer()")
-        return NullProducer()
+    logger.info("Starting KafkaProducer()")
+    return KafkaProducer(bootstrap_servers=config.bootstrap_servers)
 
 
 def emit_event(e, key):
-    producer.send(cfg.event_topic, key=key.encode("utf-8") if key else None, value=e.encode("utf-8"))
+    logger.info("Produced message to topic %s, key %s.", cfg.event_topic, key)
```
Comment (jharting) [inline tasks/__init__.py:40]: This is misleading. We should log that an event is being put into a topic only when it's actually being put into a topic

Comment (Glutexo) [reply tasks/__init__.py:40]: You’re right! 😮

**Comment by Glutexo** [comment]
Amended the log messages. Tested its workings. Thanks for the ✅ ’s!

**Comment by Glutexo** [comment]
After discussing with @jharting removed more pieces of the message queue related code. See the updated description. Re-requesting reviews.

**Comment by Glutexo** [comment]
Oh, I forgot to remove some tests. Gonna fix. :D

**Comment by Glutexo** [comment]
Fixed.

**Pull request merged**